pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -142,16 +140,19 @@ func lookupDomainInterfaceByName(domainIfaces []api.Interface, networkName strin
 }
 
 func networksToHotplugWhoseInterfacesAreNotInTheDomain(vmi *v1.VirtualMachineInstance, indexedDomainIfaces map[string]api.Interface) []v1.Network {
+	ifacesToHotplug := netvmispec.FilterIfacesForHotplug(vmi.Spec.Domain.Devices.Interfaces)
```

Comment:
I have hard time to understand the need for the changes in this function.
I would like to see the logical change needed here, but it is very hard to follow it with all the refactoring done.

I think, that we do not need all the indexing and filtering added, just a single additional condition to the predicate of `IntexInterfacesFromStatus`.
If you feel it requires refactoring, please do it in a follow up PR (doing this in this PR will delay its acceptance).

Reply:
I totally agree.

Reply:
Adding another condition to the predicate is making the code worse instead of the suggested changes.

I will post a PR for refactoring this..